app/welcome.js

```diff
@@ -6,11 +6,53 @@
  * Description:
  * Main script for initiating the welcome  page.
  */
- 
- define([ 'jquery', 'bootstrap'], function ($, bootstrap) {
 
-	$("#menu-toggle").click(function(e) {
+define(['jquery', 'app/database', 'bootstrap'], function ($, db, bootstrap) {
+
+	$("#menu-toggle").click(function (e) {
 		e.preventDefault();
 		$("#wrapper").toggleClass("toggled");
 	});
+
+	function createSidebarElements() {
+		var li1 = "<li class=\"sidebar_li\" subject_id=\"";
+    var li2 = "\" language_id=\"";
+		var li3 = "\"><a  href=\"#\"  >";
+		var li4 = "<br><span class=\"sidebar_item\" >for ";
+    var li5 = " speakers</span></a></li>\n";
+		var rows = db.getUnique('getModules', 'subject_name', []);
+		var sidebar = "";
+
+		for (var i = 0; i < rows.length; i++) {
+			sidebar += li1 + rows[i].subject_id + li2 + rows[i].language_id + li3 + rows[i].subject_name + li4 + rows[i].language_name + li5;
+		}
+
+		var container = document.getElementById("sidebar_ul");
+		container.innerHTML = sidebar;
+	}
+
+	function createDatasetsGrid(subjectid, languageid) {
+		var grid = "";
+		var rows = db.getQuery('getDatasets', [languageid, subjectid]);
+		var gridItem1 = "<div class=\"col-md-3 col-sm-6\">\n<div class=\"btn mybutton\" id=\"";
```

Comment:
If I read your code correctly you are generating the same html element multiple times, right? I cannot find a example right now, but I remember the following pattern to do this, which in my opinion is cleaner and keeps the view in the html files and the controller in the javascript files. You create your table in the html, you add one dummy row, which has all the layout you want. But you don't add any data in it, make sure to name each element. If you want to add a row to your table you retrieve the dummy row by id from the html file, you copy it and fill it with the data. This way you are way more flexible, as you can change the view in any way you want als long as the names of the dummy elements stay the same.